### ISSUE TITLE
x-form submit-on-change doesn't work with x-splade-textarea

### ISSUE THREAD

**Issue by JapSeyz** [protonemedia/laravel-splade#246]
- Laravel Version: 9.47.0
- PHP Version: 8.1.14
- Splade JS Version (npm): 1.1.7
- Splade PHP Version (composer): 1.1.7

You may use the [built-in Artisan Command](https://splade.dev/docs/upgrading) to show the installed versions of the packages:

```bash
php artisan splade:show-versions
```

### Description:

```blade
<x-splade-form :action="route('profile.update')" :default="$user" class="space-y-4" submit-on-change preserve-scroll>
        <x-splade-textarea name="description" autosize label="{{ __('Describe you and your family') }}" rows="4" />
        <x-splade-textarea name="swap_preferences" autosize label="{{ __('My house-swapping preferences are') }}" rows="4" />
    </x-splade-form>
```

As soon as a single key is pressed and released, the textarea gets disabled while it waits for the submit roundtrip. Ideally, only a blur event should trigger a save, for text fields.

**Comment by bdsumon4u** [comment]
Does it work with x-splade-select with choices?

**Comment by JapSeyz** [comment]
Apologies for the vague description. It's not that it doesn't work, more like, it works too well.

On every keypress for a text-field, the form disables the input while the form is getting submitted.

It works fine and as expected for select/choices. (as only one keypress is needed)

**Comment by pascalbaljet** [comment]
This feature was built with a select element in mind. How would you like the component to behave? More like a 'saved in the background' thing?

**Comment by iMusicJJ** [comment]
Yeah, exactly. It seems perfect for save in the background.

**Comment by pascalbaljet** [comment]
Cool, I'll work on that!

**Comment by pascalbaljet** [comment]
Not documented yet, but you may now add a `background` and `debounce` attribute to the Form Component (debounce in milliseconds):

```blade
<x-splade-form submit-on-change background debounce="500">
    <x-splade-input name="description" />

    <p v-if="form.processing">Not triggered in background</p>
    <p v-if="form.processingInBackground">Form is processing in background</p>

    <x-splade-submit />
</x-splade-form>
```

The `form.processing` prop won't do anything when it's submitting in the background, but you can still show an indicator with the `form.processingInBackground` prop.

**Comment by iMusicJJ** [comment]
Beautiful

**Comment by pascalbaljet** [comment]
I'll add something like `Splade::isBackgroundRequest()` and `Toast::info()->notOnBackgroundRequest()`. Probably have to come up with better names 😅

**Comment by JapSeyz** [comment]
Yeah sure - again it works fine as is. It just requires a very low auto-dismiss timeout, in order to prevent a flood of notifications. But it's still nice to get a visual confirmation that your details were saved.

I probably wouldn't use the `notOnBackgroundRequest` option, as it would prevent the visual cue that your changes were saved.

The optimal setup would be to limit the amount of toast w/ the same messages to only one. IMO. but that's a harder problem to tackle altogether. Therefore maybe the ability to fire an event that simply clears all toasts could also work.

Eg. on the controller you know is receiving the background request, you could clear all toasts and then simply fire the new "changes saved" message - thus ensuring only a single instance of it is shown at a time.

Again, it's only nice-to-have at best. It works fantastically w/ the low timeout now, so no worries about it :)
